cpp/benchmarks/join/join_common.hpp

```diff
@@ -58,11 +58,12 @@ struct null75_generator {
   }
 };
 
-enum class join_t { CONDITIONAL, MIXED, HASH };
+enum class join_t { CONDITIONAL, MIXED, HASH, SORT_MERGE };
 
 template <typename Key,
           bool Nullable,
-          join_t join_type = join_t::HASH,
+          join_t join_type                  = join_t::HASH,
+          cudf::null_equality compare_nulls = cudf::null_equality::UNEQUAL,
           typename state_type,
           typename Join>
```

Comment:
Absolute nit:  Where is our current thinking about grouping the type parameters together, and value parameters separately?

```suggestion
          typename state_type,
          typename Join,
          join_t join_type                  = join_t::HASH,
          bool Nullable,
          cudf::null_equality compare_nulls = cudf::null_equality::UNEQUAL>
```
Apart from grouping the type parameters together, there's coincidentally the added benefit of grouping the join-related parameters together, and the nullability ones separately. 

Is there another logical grouping here that I've missed?

Reply:
Thank you for this suggestion - I like the idea of logically grouping the template parameters. We would have to update the other join benchmarks as well once the template parameter order is changed. I'm worked on adding data type axes to our join benchmarks (https://github.com/rapidsai/cudf/pull/19281) which would further modify this function, so I would rather defer this suggestion to that PR. WDYT?

Reply:
By all means.  No worries here.